regression/goto-instrument/restrict-function-pointer-by-name-global/test.c

```diff
@@ -0,0 +1,28 @@
+#include <assert.h>
+
+int f(void)
+{
+  return 1;
+}
+
+int g(void)
+{
+  return 2;
+}
+
+int h(void)
+{
+  return 3;
+}
+
+typedef int (*fp_t)(void);
+
+fp_t fp;
+
+void main()
+{
+  int cond;
+  fp = cond ? f : g;
```

Comment:
:question: In my experience, cbmc can be smart enough to spot that `h` is unused, I guess the most precise way to test this is fixed, is have a `test-no-restriction.desc` that is the same, but check that the verification fails on the assertion :sheep:

Reply:
cbmc should always be able to figure out which functions are really called, so the verification result without using `--restrict-function-pointer` would be the same. Just the `remove_function_pointers()` pass would probably produce a superset of the functions that a user might specify via `--restrict-function-pointer`. For testing `--restrict-function-pointer`, I think it does not matter what `remove_function_pointers()` might do, as the user could also specify a superset of the functions identified by that pass.

To check that really the correct case distinction is produced, I've now also added regexes that check that precisely the correct functions are called in the transformed goto program.